File: Practice/AOJ/ITP1/ITP1_7C/itp1_7c.py
r, c = map(int, input().split())
sheet = []
c_sum = [0 for _ in range(c)]
for y in range(r):
    sheet.append(list(map(int, input().split())))
    sheet[y].append(sum(sheet[y]))
    # transposeするのはめんどくさいのでループでまわしてみる
    for x in range(c):
        c_sum[x] += sheet[y][x]
    print(" ".join(list(map(str, sheet[y]))))
c_sum.append(sum(c_sum))
print(" ".join(list(map(str, c_sum))))

File: Practice/AOJ/ITP1/ITP1_7C/itp1_7c.rb
r, c = gets.chomp.split.map(&:to_i)
sheet = []
last_culumn = Array.new(c, 0)
last_sum = 0
r.times do |y|
    line = gets.chomp.split.map(&:to_i)
    last_sum += line.sum
    line << line.sum
    sheet << line
    last_culumn = last_culumn.zip(line).map{ |a, b| a + b }
end
last_culumn << last_sum
sheet << last_culumn

sheet.each { |s| puts s.join(" ")}